src/main/kotlin/frc/robot/autonomous/ScoreState.kt

```diff
@@ -13,13 +16,22 @@ val IS_WITHIN_AUTO_SCORE_DISTANCE = {
     ) <= MAX_ALIGNMENT_DISTANCE
 }
 
-val selectedScorePose: () -> Pose2d = ScorePose.`5L`.pose
+var selectedScorePose: () -> Pose2d = ScorePose.`5L`.pose
 
-// TODO: Add implementation for actual value
-private val selectedHeightPose: () -> ScoreHeight = { ScoreHeight.L3 }
+private var selectedHeightPose: () -> ScoreHeight = { ScoreHeight.L3 }
 
-// private fun selectedScorePosePathfindingCommand(): Command =
-//    pathFindToPose(selectedScorePose.invoke())
+fun setPoseBasedOnButton(buttonID: Int): Command {
+    return runOnce({
+        selectedScorePose = {
+            buttonToPoseMap[buttonID]
+                ?: throw Exception("No pose for button $buttonID!!!")
+        }
+        Logger.recordOutput(
+            "ScoreState/SelectedScorePose",
+            selectedScorePose.invoke()
+        )
+    })
+}
```

Comment:
This is too coupled. How are you going to use this if you'll use the alignment code in auto (which you should)?
You should have a map between a button and its branch name, and another map from branch name to pose. Then you can use the second one in auto.